bindings/dotnet/openDAQ.Net/openDAQ.Net.CI.Test/OpenDAQ_CITests.cs

```diff
@@ -413,39 +445,598 @@ public void Test_0401_StreamReaderTest<TValue>(TValue _)
         Console.WriteLine($"  using signal {analogSignalNo} '{signalName}'");
 
         Console.WriteLine("OpenDAQFactory.CreateStreamReader()");
+        int loopCount    = 10;
+        nuint maxCount   = 150;
+        TValue[] samples = new TValue[maxCount];
+        FillArray(samples, (TValue)Convert.ChangeType(99999, typeof(TValue)));
+        int sleepTime          = 25;
+        nuint readSamplesCount = 0;
         //here using-sub-scope to free resources asap
-        nuint count = 500;
-        TValue defaultValue = (TValue)Convert.ChangeType(99999, typeof(TValue));
-        TValue[] samples = new TValue[count];
-        Array.Fill(samples, defaultValue);
-        nuint samplesCount = 0;
         using (var reader = OpenDAQFactory.CreateStreamReader<TValue>(signal))
         {
             Console.WriteLine($"  ValueReadType = {reader.ValueReadType}, DomainReadType = {reader.DomainReadType}");
+            Console.WriteLine($"  Reading {loopCount} times {maxCount} values (waiting {sleepTime}ms before reading)");
+
+            Stopwatch sw = new Stopwatch();
+
+            for (int loopNo = 0; loopNo < loopCount; ++loopNo)
+            {
+                //ToDo: do we get buffer overrun when we just leave reader open for too long without reading?
+
+                sw.Restart();
+
+                nuint count = (nuint)samples.Length; //reset to array size
+
+                Thread.Sleep(sleepTime); //StreamReader does not need to wait for 'AvailableCount == count' as it has 'timeoutMs'
+
+                nuint availableCount = reader.AvailableCount;
+                Debug.Print($"+++> {loopNo + 1,2}: AvailableCount={availableCount,-3}");
+
+                //read up to 'count' samples, storing the amount read into array 'samples'
+                reader.Read(samples, ref count, timeoutMs: 5000);
+                readSamplesCount += count;
+
+                sw.Stop();
+
+                string values = (count == 0) ? string.Empty : $"(0: {samples[0]:+0.000;-0.000; 0.000} ... {count - 1,3}: {samples[count - 1]:+0.000;-0.000; 0.000})";
+                Console.WriteLine($"  Loop {loopNo + 1,2} {sw.Elapsed.TotalMilliseconds,7:0.000}ms AvailableCount={availableCount,-3} read {count,3} values {values}");
+            }
+        }
+
+        Assert.That(readSamplesCount, Is.GreaterThan((nuint)0), "*** No samples received.");
+    }
+
+    [TestCase(int.MinValue)]
+    [TestCase(ulong.MinValue)]
+    [TestCase(double.MinValue)]
+    public void Test_0402_TailReaderTest<TValue>(TValue _)
```

Comment:
Mock data is used in tests in C++ and Python as I know, so you could check if there is any data damage in readers. It is also isolate your code from errors in Instance and make your tests pass faster. @JakaMohorkoDS?

Reply:
imho testing the .NET Bindings is more like an integration test, thus always testing with the reference device. 
Also, I never tried to get mocking data running in C# and I already found some reader issues this way, that have not been seen with the C++ tests (perhaps not using enough mocking data).

Reply:
according to Jaka, mocks for .NET Bindings are postponed  for now